spec/suites/map/handler/Map.TapHoldSpec.js

```diff
@@ -0,0 +1,209 @@
+		happen.once(el, L.extend({type: 'pointerdown', pointerId:1}, posNear));
+		clock.tick(100);
+		happen.once(el, {type: 'touchend', touches: [posStart]});
+		happen.once(el, L.extend({type: 'pointerup', pointerId:0}, posNear));
+		clock.tick(450);
+
+		expect(spy.notCalled).to.be.ok();
+	});
+
+	(L.Browser.pointer ? it : it.skip)('ignores events from mouse', function () {
+		happen.once(el, L.extend({type: 'pointerdown', pointerId:0, pointerType:'mouse'}, posStart));
+		clock.tick(650);
+
+		expect(spy.notCalled).to.be.ok();
+	});
+
+	it('does not conflict with native contextmenu', function () {
+		happen.once(el, {type: 'touchstart', touches: [posStart]});
+		happen.once(el, L.extend({type: 'pointerdown', pointerId:0}, posStart));
+		clock.tick(550);
+
+		happen.once(el, {type: 'contextmenu'});
+
+		clock.tick(100);
+
+		expect(spy.called).to.be.ok();
+		expect(spy.calledOnce).to.be.ok();
+		expect(spy.lastCall.args[0].originalEvent._simulated).not.to.be.ok();
+
+		// Note: depending on tapHoldDelay value it's also possible that native contextmenu may come after simulated one
+		//       and the only way to handle this gracefully - increase tapHoldDelay value.
+		//       Anyway that is edge case, as tapHold is meant for browsers where native contextmenu is not fired on touch.
+	});
+
+	it.skip('prevents native click', function () { // to be performed by hand
+		// Not valid here, as there is no way to initiate native click with fake touch
+		var clickSpy = sinon.spy();
+		map.on('click', clickSpy);
+
+		happen.once(el, {type: 'touchstart', touches: [posStart]});
+		happen.once(el, L.extend({type: 'pointerdown', pointerId:0}, posStart));
+		clock.tick(650);
+		happen.once(el, {type: 'touchend', touches: [posStart]});
+		happen.once(el, L.extend({type: 'pointerup', pointerId:0}, posNear));
+
+		expect(clickSpy.notCalled).to.be.ok();
+	});
+
+	it('allows short movements', function () {
+		happen.once(el, {type: 'touchstart', touches: [posStart]});
+		happen.once(el, L.extend({type: 'pointerdown', pointerId:0}, posStart));
+		clock.tick(550);
+
+		happen.once(el, {type: 'touchmove', touches: [posNear]});
+		happen.once(el, L.extend({type: 'pointermove', pointerId:0}, posNear));
+
+		clock.tick(100);
+
+		expect(spy.called).to.be.ok();
+	});
+
+	it('ignores long movements', function () {
+		expect(L.point(posStart.clientX, posStart.clientY).distanceTo([posFar.clientX, posFar.clientY]))
+		  .to.be.above(map.options.tapTolerance);
+
+		happen.once(el, {type: 'touchstart', touches: [posStart]});
+		happen.once(el, L.extend({type: 'pointerdown', pointerId:0}, posStart));
+		clock.tick(550);
+
+		happen.once(el, {type: 'touchmove', touches: [posFar]});
+		happen.once(el, L.extend({type: 'pointermove', pointerId:0}, posFar));
+
+		clock.tick(100);
+
+		expect(spy.notCalled).to.be.ok();
+	});
+
+	it('.originalEvent has expected properties', function () {
+		L.extend(posStart, {
+			screenX: 2,
+			screenY: 2,
+		});
```

Comment:
Should it not be `var posStartNew = L.extend(`?

Reply:
Why?

Reply:
My fault readed the docs wrong